src/NServiceBus.Extensions.Hosting/HostApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using Extensions.Hosting;
+    using Logging;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using ILoggerFactory = Microsoft.Extensions.Logging.ILoggerFactory;
+
+    /// <summary>
+    /// Extension methods to configure NServiceBus for the .NET Core generic host.
+    /// </summary>
+    public static class HostApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the host to start an NServiceBus endpoint.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="configureEndpoint"></param>
+        /// <returns></returns>
+        public static IHostApplicationBuilder UseNServiceBus(this IHostApplicationBuilder builder, Func<IHostApplicationBuilder, EndpointConfiguration> configureEndpoint)
```

Comment:
Unsure on `Func<IHostApplicationBuilder, EndpointConfiguration>`, possibly `Func<EndpointConfiguration>`, would be more appropriate given you can already access the application builder.

<img src="https://github.com/Particular/NServiceBus.Extensions.Hosting/assets/38759683/c9ef6091-e784-485e-bcc4-a5acef4ca654" width=500/>

Reply:
I agree that `Func<EndpointConfiguration>` is the better signature here given that there's no special context class to pass in.

Reply:
I've amended the changes to use the Func<out> approach, hopefully it makes for a clearer abstraction 😄